opm/simulators/linalg/cuistl/PreconditionerAdapter.hpp

```diff
@@ -80,6 +80,10 @@ class PreconditionerAdapter
             m_inputBuffer.reset(new CuVector<field_type>(v.dim()));
             m_outputBuffer.reset(new CuVector<field_type>(v.dim()));
         }
+        //TODO: to implement CuJac I added the moving of data from host to the output buffer
```

Comment:
I do not understand why this was necessary?

Reply:
I think the result depends on both v and d? When preconditioning 0 = b - Ax I need both the b and the x vector to correctly update the x.

Reply:
This has been sorted out now. It was not necessary and the apply could also be dramatically simplified for the same reason. The final version will be ready very shortly.